dspace/modules/xmlui/src/main/java/org/dspace/app/xmlui/aspect/submission/workflow/actions/processingaction/ReAuthorizationCreditActionXMLUI.java

```diff
@@ -106,16 +106,16 @@ private String submitCredit(){
                 if(concepts!=null&&concepts.length!=0){
                     AuthorityMetadataValue[] metadataValues = concepts[0].getMetadata("internal", "journal", "customerID", Item.ANY);
                     if(metadataValues!=null&&metadataValues.length>0){
+                        shoppingCart.setStatus(ShoppingCart.STATUS_COMPLETED);
+                        shoppingCart.setPaymentDate(new Date());
+                        shoppingCart.update();
+
                         try{
-                            shoppingCart.setStatus(ShoppingCart.STATUS_COMPLETED);
-                            Date date= new Date();
-                            shoppingCart.setPaymentDate(date);
-                            shoppingCart.update();
                             success = AssociationAnywhere.tallyCredit(context, metadataValues[0].value, DOIIdentifierProvider.getDoiValue(item));
                             paymentSystemService.sendPaymentApprovedEmail(context, workflowItem, shoppingCart);
                         }catch (Exception e)
                         {
-                            paymentSystemService.sendPaymentErrorEmail(context, workflowItem, shoppingCart,"problem: credit not tallied successfully. \\n \\n " + e.getMessage());
+                            paymentSystemService.sendPaymentErrorEmail(context, workflowItem, shoppingCart,"problem: credit not tallied successfully. \n \n " + e.getMessage());
                             log.error(e.getMessage());
```

Comment:
Leave the email line, just make the logging more verbose by passing the entire exception.